cmd/monaco/integrationtest/v2/references_test.go

```diff
@@ -0,0 +1,245 @@
+				assertRefParamFromTo(t, notification, profile)
+			},
+		},
+		{
+			project: "config-with-mngt-zone",
+			downloadOpts: []string{
+				"-a", "notification,alerting-profile",
+				"-s", "builtin:management-zones",
+			},
+			validate: func(t *testing.T, ctx TestContext, confsPerType project.ConfigsPerType) {
+				managementZone := findSetting(t, confsPerType, "builtin:management-zones", "zone_"+ctx.suffix, "name")
+				profile := findConfig(t, confsPerType, "alerting-profile", "profile_"+ctx.suffix)
+				notification := findConfig(t, confsPerType, "notification", "notification_"+ctx.suffix)
+
+				assertRefParamFromTo(t, profile, managementZone)
+				assertRefParamFromTo(t, notification, profile)
+			},
+		},
+	}
+
+	for _, env := range envs {
+		for _, tc := range tests {
+			testName := env + "_" + tc.project
+
+			t.Run(testName, func(t *testing.T) {
+				configFolder := "test-resources/references/"
+				manifestFile := configFolder + "manifest.yaml"
+				proj := tc.project
+
+				fs := testutils.CreateTestFileSystem()
+
+				RunIntegrationWithCleanupOnGivenFs(t, fs, configFolder, manifestFile, env, testName, func(fs afero.Fs, ctx TestContext) {
+
+					// upsert
+					cmd := runner.BuildCli(fs)
+					cmd.SetArgs([]string{"deploy", "-v", manifestFile, "--environment", env, "--project", proj})
+					err := cmd.Execute()
+					assert.Nil(t, err, "create: did not expect error")
+
+					// update just to be sure
+					cmd = runner.BuildCli(fs)
+					cmd.SetArgs([]string{"deploy", "-v", manifestFile, "--environment", env, "--project", proj})
+					err = cmd.Execute()
+					assert.Nil(t, err, "update: did not expect error")
+
+					// download
+					cmd = runner.BuildCli(fs)
+					cmd.SetArgs(append(
+						[]string{
+							"download",
+							"-v",
+							"--manifest", manifestFile,
+							"--environment", env,
+							"--project", "proj",
+							"--output-folder", "download",
+						},
+						tc.downloadOpts...))
+					err = cmd.Execute()
+					assert.Nil(t, err, "download: did not expect error")
+
+					// assert
+					mani, errs := manifest.LoadManifest(&manifest.LoaderContext{
+						Fs:           fs,
+						ManifestPath: "download/manifest.yaml",
+					})
+					assert.Empty(t, errs, "load manifest: did not expect do get error(s)")
+
+					projects, errs := project.LoadProjects(fs, project.ProjectLoaderContext{
+						KnownApis:       api.NewAPIs().GetApiNameLookup(),
+						WorkingDir:      "download",
+						Manifest:        mani,
+						ParametersSerde: config.DefaultParameterParsers,
+					})
+					assert.Empty(t, errs, "load project: did not expect do get error(s)")
+
+					projectAndEnvName := "proj_" + env // for manifest downloads proj + env name
+
+					confsPerType := findConfigs(t, projects, projectAndEnvName)
+
+					tc.validate(t, ctx, confsPerType)
+				})
+			})
+		}
+	}
+}
+
+func assertRefParamFromTo(t *testing.T, from config.Config, to config.Config) {
+	name := paramName(to.Coordinate.Type, to.Coordinate.ConfigId)
```

Comment:
relying exactly on how the parameter reference ID is formated on download seems brittle.
Couldn't you achieve virtually the same with just a 

```go
assert.Contains(t, from.References(), to.Coordinate)
```

? (leaving out the check whether it really references the ID rather than another property, but I'm not sure that level of details is even needed for this E2E test)

Reply:
Good observation, like it